diff --git a/arch/arm/boot/dts/rk3229-r329q.dts b/arch/arm/boot/dts/rk3229-r329q.dts
new file mode 100644
index 00000000..c3253d96
--- /dev/null
+++ b/arch/arm/boot/dts/rk3229-r329q.dts
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2017 Fuzhou Rockchip Electronics Co., Ltd
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *  Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+
+#include "rk3229-generic-tvbox.dtsi"
+
+/ {
+	model = "R329Q v3.x TV Box board";
+	compatible = "rockchip,rk3229-r329q", "rockchip,rk3229";
+
+	wireless-wlan {
+		compatible = "wlan-platdata";
+		rockchip,grf = <&grf>;
+		wifi_chip_type = "ssvb6501";
+		sdio_vref = <3300>;
+		WIFI,host_wake_irq = <&gpio0 28 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
+	gpio-leds {
+
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_leds>;
+
+		heartbeat {
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+			label = "r329q:red:state1";
+			linux,default-trigger = "mmc0";
+			default-state = "off";
+		};
+
+		mmc {
+			gpios = <&gpio3 21 GPIO_ACTIVE_LOW>;
+			label = "r329q:blue:state2";
+// 			linux,default-trigger = "mmc0";
+			default-state = "on";
+		};
+
+	};
+
+};
+
+/** Integration to pin controller */
+&pinctrl {
+
+	gpio {
+		gpio_leds: gpio-leds {
+			rockchip,pins = <1 7 RK_FUNC_GPIO &pcfg_pull_none>,
+					<3 21 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+};
+
+/** MMC and NAND controller and interfaces */
+
+&emmc {
+	status = "disabled";
+};
